src/chat-messages.js

```diff
@@ -0,0 +1,547 @@
+/**
+ * File: Chat Messages
+ *
+ * Maintainer:      - Sebastian Kippe <sebastian@kip.pe>
+ * Version:         - 0.5.0
+ *
+ * This module stores chat messages in daily archive files.
+ */
+
+var isNode = new Function("try {return this===global;}catch(e){return false;}");
+if (isNode) { var RemoteStorage = require("remotestoragejs"); }
+
+RemoteStorage.defineModule("chat-messages", function (privateClient, publicClient) {
+
+  /**
+   * Schema: chat-messages/daily
+   *
+   * Represents one day of chat messages
+   *
+   * Example:
+   *
+   * (start code)
+   * {
+   *   "@context": "https://kosmos.org/ns/v1",
+   *   "@id": "chat-messages/irc/freenode/channels/kosmos/",
+   *   "@type": "ChatChannel",
+   *   "name": "#kosmos",
+   *   "ircURI": "irc://irc.freenode.net/kosmos",
+   *   "today":  {
+   *     "@id": "2015/01/01",
+   *     "@type": "ChatLog",
+   *     "messageType": "InstantMessage",
+   *     "previous": "2014/12/31",
+   *     "next": "2015/01/02",
+   *     "messages": [
+   *       { "date": "2015-06-05T17:35:28.454Z", "user": "hal8000", "text": "knock knock" },
+   *       { "date": "2015-06-05T17:37:42.123Z", "user": "raucao", "text": "who's there?" },
+   *       { "date": "2015-06-05T17:55:01.235Z", "user": "hal8000", "text": "HAL" }
+   *     ]
+   *   }
+   * }
+   * (end code)
+   */
+
+  const archiveSchema = {
+    "type": "object",
+    "properties": {
+      "@context": {
+        "type": "string",
+        "default": "https://kosmos.org/ns/v1",
+        "enum": ["https://kosmos.org/ns/v1"]
+      },
+      "@id": {
+        "type": "string",
+        "required": true
+      },
+      "@type": {
+        "type": "string",
+        "default": "ChatChannel",
+        "enum": ["ChatChannel"]
```

Comment:
It's part of the SIOC vocabulary we're using. Will be announced in the @context URI later.

http://rdfs.org/sioc/spec/